delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/slowqueries/BlazegraphSlowQueryDeleter.scala

```diff
@@ -0,0 +1,41 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.slowqueries
+
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.IOUtils
+import ch.epfl.bluebrain.nexus.delta.sourcing.stream.{CompiledProjection, ExecutionStrategy, ProjectionMetadata, Supervisor}
+import fs2.Stream
+import monix.bio.Task
+
+import scala.concurrent.duration.FiniteDuration
+
+class BlazegraphSlowQueryDeleter(store: BlazegraphSlowQueryStore, deletionThreshold: FiniteDuration) {
+  def deleteOldQueries: Task[Unit] = {
+    IOUtils.instant.flatMap { now =>
+      store.removeQueriesOlderThan(now.minusMillis(deletionThreshold.toMillis))
+    }
+  }
+}
+
+object BlazegraphSlowQueryDeleter {
```

Comment:
With this it would be nice if no class was needed at all. The code located in this class doesn't really have anything to do with logging, so it's clearer if they are separate. Merging is just going to add boilerplate to the tests, as well as coupling these actions

Reply:
You can handle this with different apply methods for example one that just initiates the logger and the other that also starts the routine on top of that